bazel/scripts/run_integ_tests.sh

```diff
@@ -0,0 +1,239 @@
+    bazel build "//lte/gateway/python/integ_tests/s1aptests:test_modify_mme_config_for_sanity" --define=on_magma_test=1
+    echo "Executing..."
+    sudo "${MAGMA_ROOT}/bazel-bin/lte/gateway/python/integ_tests/s1aptests/test_modify_mme_config_for_sanity"
+    echo "Setup finished successfully."
+}
+
+teardown_extended_tests() {
+    echo "Cleaning up the environment after the extended tests."
+    echo "Building..."
+    bazel build "//lte/gateway/python/integ_tests/s1aptests:test_restore_mme_config_after_sanity" --define=on_magma_test=1
+    echo "Executing..."
+    sudo "${MAGMA_ROOT}/bazel-bin/lte/gateway/python/integ_tests/s1aptests/test_restore_mme_config_after_sanity"
+    echo "Cleanup finished successfully."
+}
+
+run_test() {
+    local TARGET=$1
+    local TARGET_PATH=${TARGET%:*}
+    local SHORT_TARGET=${TARGET#*:}
+    (
+        set -x
+        echo "BUILDING TEST: ${TARGET}"
+        bazel build "${TARGET}" --define=on_magma_test=1
+        echo "RUNNING TEST: ${TARGET}"
+        sudo "${MAGMA_ROOT}/bazel-bin/${TARGET_PATH}/${SHORT_TARGET}"
+    )
+}
+
+print_summary() {
+    local NUM_SUCCESS=$1
+    local TOTAL_TESTS=$2
+    echo "SUMMARY: ${NUM_SUCCESS}/${TOTAL_TESTS} tests were successful."
+    for TARGET in "${!TEST_RESULTS[@]}"
+    do
+        echo "  ${TARGET}: ${TEST_RESULTS[${TARGET}]}"
+    done
+}
+
+###############################################################################
+# SCRIPT SECTION
+###############################################################################
+
+TARGET_PATH="${1:-}"
+
+if [[ "${TARGET_PATH}" == "--help" ]];
+then
+    help
+    exit 0
+fi
+
+declare -a EXTENDED_TEST_TARGETS
+declare -A TEST_RESULTS
+NUM_SUCCESS=0
+NUM_RUN=1
+
+cd "${MAGMA_ROOT}"
+
+EXTENDED_TEST_SETUP="//lte/gateway/python/integ_tests/s1aptests:test_modify_mme_config_for_sanity"
+EXTENDED_TEST_TEARDOWN="//lte/gateway/python/integ_tests/s1aptests:test_restore_mme_config_after_sanity"
+SKIP_EXTENDED_SETUP_AND_TEARDOWN="false"
+
+if [[ "${TARGET_PATH}" == "--list" ]];
+then
+    list_all_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--list-extended" ]];
+then
+    list_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--list-traffic-server" ]];
+then
+    list_traffic_server_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--setup-extended" || "${TARGET_PATH}" == "${EXTENDED_TEST_SETUP}" ]];
+then
+    setup_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--teardown-extended" || "${TARGET_PATH}" == "${EXTENDED_TEST_TEARDOWN}" ]];
+then
+    teardown_extended_tests
+    exit 0
+fi
+
+if [[ "${TARGET_PATH}" == "--skip-setup-teardown-extended" ]];
+then
+    TARGET_PATH="${2:-}"
+    SKIP_EXTENDED_SETUP_AND_TEARDOWN="true"
+fi
+
+create_test_targets
+
+TOTAL_TESTS=${#EXTENDED_TEST_TARGETS[@]}
+# TODO TOTAL_TESTS=$((TOTAL_TESTS + PRECOMMIT))
+
+if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
+then
+    setup_extended_tests
+fi
+
+for TARGET in "${EXTENDED_TEST_TARGETS[@]}"
```

Comment:
I think it would be nice if there was an option to rerun failing tests for a given number of times. 

In the integ test workflow we should have at least a 3x retry on failure. If we want to simply execute the script in the workflow then it would be nice if there was at least the option to do: `$MAGMA_ROOT/bazel/scripts/run_integ_tests.sh --retry_on_failure` which could run all tests and retry 3x on failure. That should not add much complexity to the script - mainly calculating the success quota becomes somewhat more difficult. What do you think?

Reply:
In my experience with these tests just running them again does not really help, usually the problem is resolved when an other test is run. (Running test_attach_detach.py usually helps with failing tests.)

Reply:
I still think it would be a nice-to-have option, but we can always add it later on when we need it.